src/provider.ts

```diff
@@ -123,18 +125,59 @@ export class ArcanaProvider extends SafeEventEmitter {
     return url
   }
 
-  public async triggerSocialLogin(loginType: string): Promise<string> {
-    const c = await this.communication.promise
-    const url = this.getCurrentUrl()
-    const redirectUrl = await c.triggerSocialLogin(loginType, url)
-    return redirectUrl
+  triggerSocialLogin = async (loginType: string) => {
+    if (!(await this.isLoginAvailable(loginType))) {
+      throw new Error(`${loginType} login is not available`)
+    }
+
+    const redirectUrl = this.constructLoginUrl({
+      loginType,
+      appId: this.iframe.params.appId,
+    })
+
+    this.redirectTo(redirectUrl)
+    return
+  }
+
+  private redirectTo(url: string) {
+    if (url) {
+      setTimeout(() => (window.location.href = url), 50)
```

Comment:
Why do we need the setTimeout? Does this work correctly if say the network is slow or url changes take longer than 50ms?

Reply:
There has to be a certain timeout otherwise some browsers will block that redirect.